server/src/main/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapper.java

```diff
@@ -1618,6 +1620,8 @@ public boolean supportsDimension(int dims) {
             public DenseVectorIndexOptions parseIndexOptions(String fieldName, Map<String, ?> indexOptionsMap, IndexVersion indexVersion) {
                 Object mNode = indexOptionsMap.remove("m");
                 Object efConstructionNode = indexOptionsMap.remove("ef_construction");
+                Object disableOffheapCacheRescoringNode = indexOptionsMap.remove("disable_offheap_cache_rescoring");
```

Comment:
Naming is hard and I have waffled on this myself. But I think we should call this something else :(

"zero_memory_rescoring" "use_direct_io" ,etc.

And I know I said to take it out of the `rescore` path, but maybe it belongs there (LOL) because it makes it clearer what we are meaning, its ONLY for rescoring.

Reply:
It was in the rescore path, but that added complexities as `RescoreVector` is used in other scenarios where direct IO is not applicable, meaning we have to use explicit checks in code at the right points, rather than it just being an option. Or we subclass `RescoreVector`

Reply:
> We would then need to introduce a new format for every quantized variant we have.

I understand, but wouldn't every quantized format benefit from not having to load the floating point vectors for rescoring?

We need to think about this a bit.

I do agree, if we don't provide direct IO options for all quantized formats, we should not put it within the `rescore_vector` parameter options.

Reply:
`RescoreVector` is also used for search options, to override the default index options. This option won't be able to be modified for searches.